compiler/src/iree/compiler/Codegen/Common/DecomposeConvolutionToLowerDimOps.cpp

```diff
@@ -16,6 +19,106 @@ namespace mlir::iree_compiler {
 
 namespace {
 
+static bool foldHDim(linalg::DepthwiseConv2DNhwcHwcOp convOp) {
+  Value kernel = convOp.getInputs().back();
+  Value output = convOp.getOutputs().front();
+
+  auto kernelType = dyn_cast<RankedTensorType>(kernel.getType());
+  auto outputType = dyn_cast<RankedTensorType>(output.getType());
+
+  auto kernelShape = kernelType.getShape();
+  auto outputShape = outputType.getShape();
+
+  int64_t khSize = kernelShape[0];
+  int64_t ohSize = outputShape[1];
+  bool removeH = (khSize == 1 && ohSize == 1);
+
+  return removeH;
+}
+
+/// Computes a "decomposed" lowering config attribute for a conv OP
+///
+/// This method complements the patterns to decompose 2D convolutions into 1D
+/// convs. Specifically, it will update the lowering config attached to a Conv
+/// Op in a way that matches the "decomposition" patterns.
+///
+/// At the moment only Depthwise HWC convolutions are supported.
+static FailureOr<IREE::Codegen::LoweringConfigAttr>
+computeNewLCA(ArrayRef<Operation *> computeOps, MLIRContext *context) {
+
+  // 0.1 Double-check that there's only one convolution Op.
+  // TODO: Make this hook work with multiple conv Ops
+  assert(llvm::count_if(computeOps,
+                        [](Operation *op) {
+                          return isa<linalg::ConvolutionOpInterface>(op);
+                        }) == 1 &&
+         "Exactly 1 Linalg Conv Op is expected");
+
+  // 0.2 ATM only 2D depthwise HWC convs are supported.
+  // TODO: Add support for other convs
+  bool seen2DDepthwiseConvHWC = false;
+  linalg::DepthwiseConv2DNhwcHwcOp convOp;
+  for (auto op : computeOps) {
+    if (isa<linalg::DepthwiseConv2DNhwcHwcOp>(op)) {
+      seen2DDepthwiseConvHWC = true;
+      convOp = cast<linalg::DepthwiseConv2DNhwcHwcOp>(op);
+      break;
+    }
+  }
+  if (!seen2DDepthwiseConvHWC)
+    return failure();
```

Comment:
Given that we have an assertion above and we only check one operation type here, we can simplify it to

```cpp
auto convOp = cast<linalg::DepthwiseConv2DNhwcHwcOp>(op);
if (!convOp) {
  return failure();
}
```

side note: we can use TypeSwitch when we want to support other convs.

Reply:
oops, I misread the logic. The updated version looks good to me. Thanks for making it better